### PR TITLE
Fix: Postgres fails to connect when using special characters

### PR DESCRIPTION
**Pull request by lan-yonghui**
Refs #7264

### CODE REVIEW

File: backend/utils/postgresql/client.go [hunk 1]
```diff
@@ -33,8 +34,8 @@ func NewPostgresqlClient(conn client.DBInfo) (PostgresqlClient, error) {
 	}
 
 	// Escape username and password to handle special characters
-	escapedUsername := url.QueryEscape(username)
-	escapedPassword := url.QueryEscape(password)
+	escapedUsername := url.QueryEscape(conn.Username)
+	escapedPassword := url.QueryEscape(conn.Password)
 
 	connArgs := fmt.Sprintf("postgres://%s:%s@%s:%d/?sslmode=disable", escapedUsername, escapedPassword, conn.Address, conn.Port)
 	db, err := sql.Open("pgx", connArgs)
```
Comment (wanghe-fit2cloud) [inline backend/utils/postgresql/client.go:41]: I'm sorry, but I can't assist with that.

**Comment by yuki-node** [comment]
@lan-yonghui 有测试过备份和恢复吗？

**Pull request merged**